common/rpc/factory.go

```diff
@@ -47,19 +48,26 @@ type Factory struct {
 // NewFactory builds a new rpc.Factory
 func NewFactory(logger log.Logger, p Params) *Factory {
 	inbounds := yarpc.Inbounds{}
+
+	transportOptions := []tchannel.TransportOption{
+		tchannel.ServiceName(p.ServiceName),
+		tchannel.ListenAddr(p.TChannelAddress),
+	}
+
+	if p.InboundTLS != nil {
+		transportOptions = append(transportOptions,
+			tchannel.InboundTLSConfiguration(p.InboundTLS),
+			tchannel.InboundTLSMode(yarpctls.Enforced),
+		)
+	}
```

Comment:
do you know if TLS is (probably silently) ignored/downgraded if it's not enabled?  kinda seems like it shouldn't be anyway...

more generally though, we probably can't enable and enforce in the same step.
validation's useful, and e.g. if this is used inter-Cadence-service, it'll mean we can't deploy because non-TLS won't be able to communicate with the new TLS instances, effectively making a split brain.

Reply:
yep, good points. Will update this PR to include option for setting TLS mode. https://pkg.go.dev/go.uber.org/yarpc@v1.70.3/api/transport/tls#Mode